projects/igniteui-angular/src/lib/directives/tooltip/tooltip.directive.ts

```diff
@@ -110,6 +120,19 @@ export class IgxTooltipDirective extends IgxToggleDirective {
         @Optional() navigationService: IgxNavigationService) {
         // D.P. constructor duplication due to es6 compilation, might be obsolete in the future
         super(elementRef, cdr, overlayService, navigationService);
+
+        this.opening.pipe(first(), takeUntil(this._destroy$)).subscribe(() => {
+            document.addEventListener('touchstart', this.onDocumentTouchStart, { passive: true });
```

Comment:
I don't like the idea of passing around the onDocumentTouchStart handler.

And I think that the current code is wrong.
1st target opens the tooltip -> adds event listener  (no one removes it)
...
Nth target opens the tooltip -> add event listener

When the tooltip gets destroyed only the last listener is detached.
And also, there's the missed opportunity of cleaning up after each target was left.

Reply:
Actually, adding the event listener in the `opening` event of `this.target` (tooltip) will add it 1000 times, while the current approach adds it once. Even if the event listeners are removed in the `closing` event, the `onDocumentTouchStart` method will be called 1000 times before that. This also seems to throw the `[Violation] 'touchstart' handler took 1102ms` warning.

![documentTouchStart](https://github.com/user-attachments/assets/0db29e89-8580-477a-9c74-4c3c85b370d3)


I'm not sure that adding/removing the document:touchStart event should be handled in the tooltipTarget since there are multiple tooltipTargets (purple squares) with a single tooltip.

Reply:
It can be checked that only the specific target opens it.

I don't think that you have addressed the issue of leaking listeners.
You attach on each open event, but you only detach on destroy.